File: rfidreader.server/src/main/kotlin/com/rfidreader/services/disciplines/DisciplineService.kt
package com.rfidreader.services.disciplines

import com.rfidreader.services.disciplines.models.DisciplineDto
import com.rfidreader.services.disciplines.models.NewDiscipline
import com.rfidreader.services.disciplines.models.UpdateDiscipline
import org.springframework.stereotype.Service

interface DisciplineService {
    fun addDiscipline(newDiscipline: NewDiscipline): Unit
    fun removeDiscipline(id: Long): Unit
    fun updateDiscipline(discipline: UpdateDiscipline): Unit

    fun getAllDisciplines(): List<DisciplineDto>
    fun getDisciplineByLecturer(id: Long): List<DisciplineDto>
}

File: rfidreader.desktop/src/renderer/src/services/DisciplineService.ts
import { IDisciplineInfo, INewDiscipline, IUpdateDiscipline } from "@core/models/discipline";
import $api from "@core/utils/api";
import { AxiosResponse } from "axios";

class DisciplineService {
    public async getAllDisciplines(): Promise<AxiosResponse<IDisciplineInfo[]>> {
        return await $api.get(`/discipline/getAll`)
    }
    public async addDiscipline(discipline: INewDiscipline): Promise<AxiosResponse> {
        return await $api.post(`/discipline/add`, discipline)
    }   
    public async removeDiscipline(id: number): Promise<AxiosResponse> {
        return await $api.delete(`/discipline/remove/${id}`)
    }
    public async updateDiscipline(group: IUpdateDiscipline): Promise<AxiosResponse> {
        return await $api.put(`/discipline/update`, group)
    }
}
export const disciplineService = new DisciplineService()